src/gui/properties/peersadditiondlg.cpp

```diff
@@ -0,0 +1,111 @@
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ *
+ * Contact : chris@qbittorrent.org
+ */
+
+#include <QMessageBox>
+#include <QHostAddress>
+
+#include "peersadditiondlg.h"
+
+PeersAdditionDlg::PeersAdditionDlg(QWidget *parent)
+    : QDialog(parent)
+{
+    setupUi(this);
+    connect(buttonBox, SIGNAL(accepted()), this, SLOT(validateInput()));
+
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+    label_format->hide();
+    peers_txt->setPlaceholderText("Format: IPv4:port / [IPv6]:port");
+#endif
+}
+
+QList<BitTorrent::PeerAddress> PeersAdditionDlg::askForPeersEndpoints()
+{
+    PeersAdditionDlg dlg;
+    dlg.exec();
+    return dlg.m_peersList;
+}
+
+void PeersAdditionDlg::validateInput()
+{
+    if (peers_txt->toPlainText().trimmed().isEmpty()) {
+        QMessageBox::warning(this, tr("No peer entered"),
+                    tr("Please type at least one peer."),
+                    QMessageBox::Ok);
+        return;
+    }
+
+    bool allValid = true;
+    QStringList peers = peers_txt->toPlainText().trimmed().split("\n");
+    foreach (const QString &peer, peers) {
+        BitTorrent::PeerAddress addr = parsePeer(peer);
+        if (!addr.ip.isNull()) {
+            m_peersList.append(addr);
+        } else {
+	    QMessageBox::warning(this, tr("Invalid peer"),
+		    tr("The peer %1 is invalid.").arg(peer),
+		    QMessageBox::Ok);
+	    allValid = false;
+	    break;
+        }
+    }
+
+    if (allValid == false)
+        m_peersList.clear();
+    else
+        accept();
+}
+
+BitTorrent::PeerAddress PeersAdditionDlg::parsePeer(QString peer)
+{
+    BitTorrent::PeerAddress addr;
+    QString strIP, strPort;
+
+    if (peer.indexOf("[") != -1 && peer.indexOf("]:") != -1) { // IPv6
+        strIP = peer.split("]:")[0].remove(QChar('['));
+        strPort = peer.split("]:")[1];
+    } else if (peer.indexOf(":") != -1) { // IPv4
```

Comment:
I made the port always compulsory (as it's in the format label) because there isn't a default port. Each client has his own default port and most of them chose the port randomly.